nimbleclient/v1/restclient.py

```diff
@@ -266,10 +294,16 @@ def delete(self, endpoint):
             raise ConnectionError(f"Error communicating with {self.hostname}")
 
     def put(self, endpoint, **payload):
-        """Wrapper for PUT requests"""
+        """Wrapper for PUT requests."""
+
+        # Translate metadata dict to REST key-value pairs.
+        # Ex: {'k1': 'k1', 'k2': 'v2'} ==> [{'key': 'k1', 'value': 'v1'}, {'key': 'k2', 'value': 'v2'}]
+        if 'metadata' in payload:
+            payload['metadata'] = [{'key': key, 'value': payload['metadata'][key]} for key in payload['metadata']]
 
         try:
             while 1:
+                logging.debug("Payload: %s", json.dumps(payload, indent=4))
```

Comment:
The payload can contain sensitive data that we shouldn't log. I would remove this statement.

Reply:
Agreed, removing all those logs